src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -14415,6 +14415,14 @@ static void M()
                 symbolValidator: module =>
                 {
                     AssertEx.Equal(new[] { "<>y__InlineArray2", "<>y__InlineArray3" }, getInlineArrayTypeNames(module));
+
+                    foreach (var inlineArrayType in getInlineArrayTypes(module))
+                    {
+                        Assert.Equal(System.Runtime.InteropServices.LayoutKind.Sequential, inlineArrayType.Layout.Kind);
+                        Assert.Single(inlineArrayType.GetAttributes(), isCompilerGeneratedAttribute);
+                        var field = Assert.Single(inlineArrayType.GetMembers().OfType<FieldSymbol>());
+                        Assert.Empty(field.GetAttributes().Where(isCompilerGeneratedAttribute));
```

Comment:
I've put together [a helper function](https://github.com/dotnet/roslyn/blob/935befe3518e902e58508e591a497c1ed7cbae4b/src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs#L12878-L12884) to verify a compiler-generated type in one of my previous PRs. Maybe we should extract it to some base test type and reuse here since it seems that "missing a `[CompilerGenerated]` attribute" is a somewhat common flaw

Reply:
Yes, let's consider making `verifyCompilerGeneratedType()` a static method in this test class, and using that method here.